src/instructlab/model/chat.py

```diff
@@ -789,18 +789,47 @@ def chat_cli(
     sys_prompt = CONTEXTS.get(context, "default")(get_model_arch(pathlib.Path(model)))
     loaded["messages"] = [{"role": "system", "content": sys_prompt}]
 
+    retriever: DocumentStoreRetriever | None = None
+
     # Instantiate retriever if RAG is enabled
     if rag_enabled:
         logger.debug("RAG enabled for chat; initializing retriever")
-        retriever: DocumentStoreRetriever | None = create_document_retriever(
-            document_store_uri=document_store_uri,
-            document_store_collection_name=collection_name,
-            top_k=top_k,
-            embedding_model_path=embedding_model_path,
-        )
+        if not os.path.exists(embedding_model_path) or not os.access(
+            embedding_model_path, os.R_OK
+        ):
+            logger.error(
+                f"Embedding model is not found: {embedding_model_path}\n"
+                "  This is typically addressed by running: ilab model download -rp <model-location>\n"
+                "  Where <model-location> is a location for your model, e.g. ibm-granite/granite-embedding-125m-english"
```

Comment:
I think we should avoid referencing a specific model or CLI command here. If the default embedding model changes, this message will become possibly misleading. It will be easy to forget to update error strings (or not know about them) in line with future changes. This is also not a CLI class, so this error message may end up getting unwittingly refactored into being returned from a REST API in the future.

I would suggest the strategy described in [this ADR](https://github.com/instructlab/dev-docs/pull/186) - throw a custom exception here, catch that in the CLI layer and print there. There's an example of this pattern [in my PR here](https://github.com/instructlab/instructlab/pull/3056).

Reply:
Does that mean you want to terminate the chat in that case?  I had assumed this was different because chat is interactive (unlike most other CLI commands).

Reply:
I replaced the hard-coded reference to `ibm-granite/granite-embedding-125m-english` with an actual reference to `{defaults.DEFAULTS.GRANITE_EMBEDDING_MODEL_NAME}`.  Let's discuss the other changes you are requesting before proceeding.

Reply:
For anyone following along here: Anastas and I met and decided that we will terminate the chat when the RAG initialization fails.  So I will proceed with his original suggestion: add new custom exceptions for failing to load the embedding model and failing to load the vector DB.

Reply:
Done!  @anastasds , can you re-review?